sympy/core/mul.py

```diff
@@ -195,6 +195,10 @@ def flatten(cls, seq):
                                 bargs.insert(0, ar)
                             bargs = [Add._from_args(bargs)]
                             rv = bargs, [], None
+            if a.is_commutative == True and b.is_commutative == True:
+                if ((a.is_finite == False and b.is_zero == True) or
+                        (b.is_finite == False and a.is_zero == True)):
+                    rv = [S.NaN], [], None
```

Comment:
Didn't get you. What it should be instead?
On Mar 21, 2015 8:43 PM, "Joachim Durchholz" <notifications@github.com>
wrote:

> In sympy/core/mul.py
> <https://github.com/sympy/sympy/pull/9179#discussion_r26892673>:
>
> > @@ -195,6 +195,10 @@ def flatten(cls, seq):
> >                                  bargs.insert(0, ar)
> >                              bargs = [Add._from_args(bargs)]
> >                              rv = bargs, [], None
> > +            if a.is_commutative == True and b.is_commutative == True:
> > +                if ((a.is_finite == False and b.is_zero == True) or
> > +                        (b.is_finite == False and a.is_zero == True)):
> > +                    rv = [S.NaN], [], None
>
> Why == True?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9179/files#r26892673>.
>